### ISSUE TITLE
Improvement: use shader name in source-code-filenames instead of only hash

### ISSUE THREAD

**Issue by teadrinker** [GabeRundlett/desktop-shadertoy#12]
When opening multiple projects/shaders and specially different revisions, it's confusing in the text editor unless you memorize the hashes.

On the site, both hashes and names are unique, however locally, neither name or hash is unique (if I understand correctly)

**Comment by GabeRundlett** [comment]
You're absolutely right. The way things are hashed currently is really bad and I don't yet know a solution. The files however are not meant to be interacted with via the filesystem... I wanted you to only ever open/save/import/export via the packed json file.
